align.go

```diff
@@ -0,0 +1,115 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package plot
+
+import (
+	"math"
+
+	"github.com/gonum/plot/vg"
+	"github.com/gonum/plot/vg/draw"
+)
+
+// Align returns a two-dimensional row-major array of Canvases which will
+// produce tiled plots with DataCanvases that are evenly sized and spaced.
+// The arguments to the function are a two-dimensional row-major array
+// of plots, a tile configuration, and the canvas to which the tiled
+// plots are to be drawn.
+func Align(plots [][]*Plot, t draw.Tiles, dc draw.Canvas) [][]draw.Canvas {
```

Comment:
what about returning a, say, `Canvases` type instead of this slice of slices?
this way, we could attach methods to it (`At(i,j int)` comes to mind) and implement e.g. `Thumbnailer`, `Plotter`, ...

one could then create these type of plots more easily:
- http://www.cookbook-r.com/Graphs/Scatterplot/
- http://www.cookbook-r.com/Graphs/Scatterplot/figure/unnamed-chunk-4-2.png

that's more or less what I did here:
- https://github.com/go-hep/hep/tree/master/hplot#tiles-of-1d-histograms
- https://github.com/go-hep/hep/blob/master/hplot/tiledplot.go#L22